src/Articulate/MetaWeblog/ArticulateMetaWeblogProvider.cs

```diff
@@ -0,0 +1,512 @@
+                            firstImage = mediaFileSystemPath;
+                        }
+                        return " src=\"" + mediaFileSystemPath + "\"";
+                    }
+                    return null;
+                });
+
+                var imagesProcessed = 0;
+
+                // Now ensure all anchors have the custom class
+                // and the media file system path is re-updated as per above
+                contentToSave = _mediaHref.Replace(contentToSave, match =>
+                {
+                    if (match.Groups.Count == 2)
+                    {
+                        var relativePath = match.Groups[1].Value;
+                        var mediaFileSystemPath = _mediaFileManager.FileSystem.GetUrl(relativePath);
+
+                        var href = " href=\"" +
+                               mediaFileSystemPath +
+                               "\" class=\"a-image-" + imagesProcessed + "\" ";
+
+                        imagesProcessed++;
+
+                        return href;
+                    }
+                    return null;
+                });
+
+                content.SetInvariantOrDefaultCultureValue("richText", contentToSave, contentType, _localizationService);
+                if (extractFirstImageAsProperty
+                && content.HasProperty("postImage")
+                    && !firstImage.IsNullOrWhiteSpace())
+                {
+                    var configuration = _dataTypeService.GetDataType(content.Properties["postImage"].PropertyType.DataTypeId).ConfigurationAs<ImageCropperConfiguration>();
+                    var crops = configuration?.Crops ?? Array.Empty<ImageCropperConfiguration.Crop>();
+
+                    var imageCropValue = new ImageCropperValue
+                    {
+                        Src = firstImage,
+                        Crops = crops.Select(x => new ImageCropperValue.ImageCropperCrop
+                        {
+                            Alias = x.Alias,
+                            Height = x.Height,
+                            Width = x.Width
+                        }).ToList()
+                    };
+
+                    content.SetInvariantOrDefaultCultureValue(
+                        "postImage",
+                        JsonConvert.SerializeObject(imageCropValue),
+                        contentType,
+                        _localizationService);
+                }
+            }
+
+            if (!post.link.IsNullOrWhiteSpace())
+            {
+                content.SetInvariantOrDefaultCultureValue("umbracoUrlName", post.link, contentType, _localizationService);
+            }
+            if (!post.mt_excerpt.IsNullOrWhiteSpace())
+            {
+                content.SetInvariantOrDefaultCultureValue("excerpt", post.mt_excerpt, contentType, _localizationService);
+            }
+
+            // TODO: Investigate why this not supported ?!
```

Comment:
Was this an existing TODO or a new one? If it's new, do we have a new task for this?

Reply:
Comments will be supported once this PR is merged https://github.com/shawnwildermuth/MetaWeblog/pull/16